week-5/C-webchat/exercise-1.js

```diff
@@ -34,4 +34,24 @@ When you open index.html in your browser, it should display the existing message
 */
 
 
-// Write your code here
\ No newline at end of file
+// Write your code here
+
+var messageList = document.querySelector("#message-list"); // select a message-list div
+
+setInterval(function() {
+	fetch("https://codeyourfuture.herokuapp.com/api/messages")
+		.then(function(response) {
+			return response.json(); // changed to json for better look in the page
+		})
+		.then(function(messages) {
+            messageList.innerHTML = ""; // clear the message box every 2 seconds
+			// iterate through an array of objects and store an array of html text
+			var messageText = messages.map(m => m.content + " <span style='float:right'>" + m.datetime + "</span>");
+			// iterate through an array of html text and put it inside paragraphs on the page
+			messageText.map(m => {
+				var p = document.createElement("p");
+				p.innerHTML = m;
+				messageList.appendChild(p);
+			});
+		});
+}, 2000);
```

Comment:
Using an inline function with `setInterval` means that the user has to wait for 2 seconds before the information appears. It's a good idea to extract a named function, e.g. `loadMessages`, then you can call it when the page loads and also pass it to `setInterval` to refresh every 2 seconds:

    loadMessages();
    setInterval(loadMessages, 2000);